deps/wazuh_testing/wazuh_testing/scripts/statistical_data_analyzer.py

```diff
@@ -0,0 +1,339 @@
+import argparse
+import os
+import sys
+import pandas as pd
+from prettytable import PrettyTable
+from scipy.stats import ttest_ind, levene, f_oneway
+
+STATS_MAPPING = {
+    'cpu': 'CPU(%)',
+    'memory': 'RSS(KB)',
+    'virtual_memory': 'VMS(KB)',
+    'file_descriptor': 'FD',
+    'read_ops': 'Read_Ops',
+    'write_ops': 'Write_Ops',
+    'disk_read': 'Disk_Read(B)',
+    'disk_written': 'Disk_Written(B)',    
+    'disk_usage': 'Disk(%)',
+    'uss': 'USS(KB)',
+    'pss': 'PSS(KB)',
+    'swap': 'SWAP(KB)',
+}
+
+STATISTICS_LIST = ['Mean', 'Median', 'Max value', 'Min value', 'Standard deviation', 'Variance']
+
+def get_parameters():
+    """Get and process script parameters.
+    
+    Returns:
+        argparse.Namespace: script parameters.
+    """
+    arg_parser = argparse.ArgumentParser()
+
+    arg_parser.add_argument('-b', '--baseline', metavar='<file>', type=str, action='store',
+                            help='Baseline file to compare', required=True, dest='baseline_data')
+
+    arg_parser.add_argument('-f', '--file', metavar='<file>', type=str, action='store',
+                            help='Data file to compare', required=True, dest='data_source')
+    
+    return arg_parser.parse_args()
+
+
+def validate_parameters(parameters, datasource):
+    """Validate the input parameters
+
+    Args:
+        parameters (argparse.Namespace): script parameters.
+        datframe: data frame obtained from the data source file
+    """
+    # Check that the file exists
+    if not os.path.exists(parameters.baseline_data) or not os.path.exists(parameters.data_source):
+        print(f"The files provided do not exist")
+        sys.exit(1)
+    
+    # Check that the source file has more than 0 rows
+    dataframe = load_dataframe(datasource)
+    data_length = len(dataframe)    
+    if data_length == 0:
+        print(f"The source '{dataframe}' has not data rows or it has not CSV format")
+        sys.exit(1)
+
+
+def load_dataframe(datasource):
+    """Read the CSV and convert it to dataframe
+
+    Args:
+        datasource: data source file path.
+
+    Returns:
+        DataFrame: dataframe object.
+    """
+    return pd.read_csv(datasource)
+
+
+def calculate_basic_statistics(dataframe):
```

Comment:
There is no reference to the expected format of the CSV. Module usage should be clear reading module docstrings or specifics methods

Reply:
Done in https://github.com/wazuh/wazuh-qa/pull/5622/commits/0dcb61333d61863972d8521e0fb586ce9c779a3b